core/src/main/java/io/questdb/griffin/engine/functions/groupby/RegressionSlopeFunctionFactory.java

```diff
@@ -154,23 +152,37 @@ public boolean isThreadSafe() {
 
         @Override
         public void merge(MapValue destValue, MapValue srcValue) {
-            double sum_x = srcValue.getDouble(valueIndex);
-            double sum_y = srcValue.getDouble(valueIndex + 1);
-            double sum_xy = srcValue.getDouble(valueIndex + 2);
-            double sum_x_squared = srcValue.getDouble(valueIndex + 3);
-            long count = srcValue.getLong(valueIndex + 4);
-
-            double sum_x2 = destValue.getDouble(valueIndex);
-            double sum_y2 = destValue.getDouble(valueIndex + 1);
-            double sum_xy2 = destValue.getDouble(valueIndex + 2);
-            double sum_x2_squared = destValue.getDouble(valueIndex + 3);
-            long count2 = destValue.getLong(valueIndex + 4);
-
-            destValue.putDouble(valueIndex, sum_x + sum_x2);
-            destValue.putDouble(valueIndex + 1, sum_y + sum_y2);
-            destValue.putDouble(valueIndex + 2, sum_xy + sum_xy2);
-            destValue.putDouble(valueIndex + 3, sum_x_squared + sum_x2_squared);
-            destValue.putLong(valueIndex + 4, count + count2);
+            double srcMeanY = srcValue.getDouble(valueIndex);
+            double srcMeanX = srcValue.getDouble(valueIndex + 1);
+            double srcSumXY = srcValue.getDouble(valueIndex + 2);
+            double srcSumX = srcValue.getDouble(valueIndex + 3);
+            long srcCount = srcValue.getLong(valueIndex + 4);
+
+            double destMeanY = destValue.getDouble(valueIndex);
+            double destMeanX = destValue.getDouble(valueIndex + 1);
+            double destSumXY = destValue.getDouble(valueIndex + 2);
+            double destSumX = destValue.getDouble(valueIndex + 3);
+            long destCount = destValue.getLong(valueIndex + 4);
+
+            long mergedCount = srcCount + destCount;
+            double deltaY = destMeanY - srcMeanY;
+            double deltaX = destMeanX - srcMeanX;
+
+            // This is only valid when countA is much larger than countB.
+            // If both are large and similar sizes, delta is not scaled down.
+            // double mergedMean = srcMean + delta * ((double) destCount / mergedCount);
+
+            // So we use this instead:
+            double mergedMeanY = (srcCount * srcMeanY + destCount * destMeanY) / mergedCount;
+            double mergedMeanX = (srcCount * srcMeanX + destCount * destMeanX) / mergedCount;
+            double mergedSumXY = srcSumXY + destSumXY + (deltaY * deltaX) * ((double) (srcCount * destCount) / mergedCount);
```

Comment:
the long multiplication will overflow with incorrect value if you merge past 10Bn and 1Bn values. The `srcCount` should be cast to `double` before the multiplication.

Additionally, the cluster 

```java
(deltaY * deltaX) * ((double) (srcCount * destCount) / mergedCount)
```

can be pre-computed and reused in both `mergedSumXY` and `mergedSumX`. I would not trust Java to optimise that.

Reply:
ignore the extraction bit, i noticed that those are not the same calculations

Reply:
I would extract:

```java
double x = (((double) srcCount * destCount) / mergedCount)
```